src/Core/IdentityServer/ResourceOwnerPasswordValidator.cs

```diff
@@ -55,6 +57,14 @@ protected async override Task<(User, bool)> ValidateContextAsync(ResourceOwnerPa
                 return (null, false);
             }
 
+            // Uncomment whenever we want to require the `auth-email` header
+            //
+            //if (!_currentContext.HttpContext.Request.Headers.ContainsKey("Auth-Email") ||
+            //    _currentContext.HttpContext.Request.Headers["Auth-Email"] != context.UserName)
```

Comment:
Does this need to compare lower invariants?

Reply:
No, it is not case-sensitive.

Reply:
In that case, I believe you need something like 
```c#
String.Equals(_currentContext.HttpContext.Request.Headers["Auth-Email"], context.UserName, StringComparison.InvariantCultureIgnoreCase)
```

Reply:
I think you misread my last comment. The case is not sensitive here.

Reply:
for everyone else:
Kyle thought I meant `Auth-Email` but meant the value of `_currentContext.HttpContext.Request.Headers.ContainsKey("Auth-Email")`.

Case is considered here since the clients will be sending the same string back used to auth anyway